docs/sql-ref-literals.md

```diff
@@ -51,6 +51,20 @@ A string literal is used to specify a character string value.
 
     Case insensitive, indicates `RAW`. If a string literal starts with `r` prefix, neither special characters nor unicode characters are escaped by `\`.
 
+The following escape sequences are recognized in regular string literals (without the `r` prefix), and replaced according to the following rules:
+- `\0` -> `\u0000`, unicode character with the code 0;
+- `\'` -> `'`, skip the slash before `'`;
+- `\"` -> `"`, skip the slash before `"`;
```

Comment:
How is this different from ``` `\<other char>` -> `<other char>` ```? I think the key difference is, here the escaping happens, while `\<other char>` is kind of a fallback for invalid escaping. Shall we make this clearer?

Reply:
I think we can remove this cases because they duplicate the default case. In the PR, I removed them: https://github.com/apache/spark/pull/43170

Reply:
I removed the unnecessary cases: https://github.com/apache/spark/pull/43170